app/ui/legacy/src/main/java/com/fsck/k9/activity/MessageListActivityAppearance.kt

```diff
@@ -0,0 +1,67 @@
+package com.fsck.k9.activity
+
+import com.fsck.k9.Account
+import com.fsck.k9.FontSizes
+import com.fsck.k9.K9
+
+class MessageListActivityAppearance(accounts: List<Account>) {
+    val appTheme = K9.appTheme
+    val k9Language = K9.k9Language
+    val isHideSpecialAccounts = K9.isHideSpecialAccounts
+    val isShowMessageListStars = K9.isShowMessageListStars
+    val isShowCorrespondentNames = K9.isShowCorrespondentNames
+    val isMessageListSenderAboveSubject = K9.isMessageListSenderAboveSubject
+    val isShowContactName = K9.isShowContactName
+    val isChangeContactNameColor = K9.isChangeContactNameColor
+    val isShowContactPicture = K9.isShowContactPicture
+    val isColorizeMissingContactPictures = K9.isColorizeMissingContactPictures
+    val isUseBackgroundAsUnreadIndicator = K9.isUseBackgroundAsUnreadIndicator
+    val contactNameColor = K9.contactNameColor
+    val messageViewTheme = K9.messageViewTheme
+    val messageListPreviewLines = K9.messageListPreviewLines
+    val splitViewMode = K9.splitViewMode
+    val fontSizes = K9.fontSizes
+    val chipColor = accounts.map { it.chipColor }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is MessageListActivityAppearance) return false
+
+        if (appTheme != other.appTheme) return false
+        if (k9Language != other.k9Language) return false
+        if (isHideSpecialAccounts != other.isHideSpecialAccounts) return false
+        if (isShowMessageListStars != other.isShowMessageListStars) return false
+        if (isShowCorrespondentNames != other.isShowCorrespondentNames) return false
+        if (isMessageListSenderAboveSubject != other.isMessageListSenderAboveSubject) return false
+        if (isShowContactName != other.isShowContactName) return false
+        if (isChangeContactNameColor != other.isChangeContactNameColor) return false
+        if (isShowContactPicture != other.isShowContactPicture) return false
+        if (isColorizeMissingContactPictures != other.isColorizeMissingContactPictures) return false
+        if (isUseBackgroundAsUnreadIndicator != other.isUseBackgroundAsUnreadIndicator) return false
+        if (contactNameColor != other.contactNameColor) return false
+        if (messageViewTheme != other.messageViewTheme) return false
+        if (messageListPreviewLines != other.messageListPreviewLines) return false
+        if (splitViewMode != other.splitViewMode) return false
+        if (chipColor != other.chipColor) return false
+        if (!areFontSizesEquals(fontSizes, other.fontSizes)) return false
+
+        return true
+    }
+
+    private fun areFontSizesEquals(fontSizes1: FontSizes, fontSizes2: FontSizes) = when {
```

Comment:
`FontSizes` is a mutable class just like `Account`. It has the same lifetime, i.e. a given `Account` instance only every references the same `FontSizes` instance. So we need to copy the values, not just hold on to the reference. (See suggested code snippet above).

Right now this method will always return `true` because `fontSizes1 === fontSizes2`.